### PR TITLE
Add `ops.random.alpha_dropout` and `layers.AlphaDropout`

### PR DESCRIPTION
**Pull request by james77777778**
Related to #18467 

This PR should benefit the users who want to apply the self-normalizing property to their models.

The implementation is taken from:
https://github.com/keras-team/keras/blob/v2.14.0/keras/layers/regularization/alpha_dropout.py#L28-L104

### CODE REVIEW

File: keras/backend/jax/random.py [hunk 1]
```diff
@@ -81,6 +82,24 @@ def dropout(inputs, rate, noise_shape=None, seed=None):
     )
 
 
+def alpha_dropout(inputs, rate, noise_shape=None, seed=None):
+    noise_shape = _get_concrete_noise_shape(inputs, noise_shape)
+    alpha = 1.6732632423543772848170429916717
```
Comment (AakashKumarNain) [inline keras/backend/jax/random.py:87]: Any specific reason for having more than 6-7 decimal points?

Comment (fchollet) [reply keras/backend/jax/random.py:87]: In practice it will be cast to float32 so some precision will be lost. The number above is just taken from the paper.

File: keras/layers/regularization/alpha_dropout.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+        self.rate = rate
+        self.seed = seed
+        self.noise_shape = noise_shape
+        self.seed_generator = backend.random.SeedGenerator(seed)
+        self.supports_masking = True
+        self.built = True
+
+    def call(self, inputs, training=False):
+        if training and self.rate > 0:
+            return backend.random.alpha_dropout(
```
Comment (fchollet) [inline keras/layers/regularization/alpha_dropout.py:51]: We can implement the layer in terms of backend ops and `random.uniform`. There is no real need for the `alpha_dropout` op -- only the layer would get used (even then, it's fairly niche usage).

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! Ok to add the layer -- but let's drop the op.

File: keras/layers/regularization/alpha_dropout.py (hunk 2 above)
Comment (james77777778) [reply keras/layers/regularization/alpha_dropout.py]: Sure, I've made the changes.

**Review by fchollet — APPROVED** [review]
Thank you for the contribution!
